lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -99,17 +102,19 @@ def validate_secret_spec(secret_name, spec)
     end
 
     def create_or_update_secret(secret_name, secret_type, data)
-      secret_yaml = generate_secret_yaml(secret_name, secret_type, data)
       msg = secret_exists?(secret_name) ? "Updating secret #{secret_name}" : "Creating secret #{secret_name}"
       KubernetesDeploy.logger.info(msg)
 
+      secret_yaml = generate_secret_yaml(secret_name, secret_type, data)
       file = Tempfile.new(secret_name)
       file.write(secret_yaml)
       file.close
 
       out, err, st = run_kubectl("apply", "--filename=#{file.path}")
       KubernetesDeploy.logger.debug(out)
       raise EjsonSecretError, err unless st.success?
+    ensure
+      file.unlink if file
```

Comment:
I wonder if this can actually error out in some circumstances, I'd be fine to rely on Tempfile to take care of this. But I'm fine either way